app/assets/javascripts/discourse/app/components/add-topic-status-classes.js

```diff
@@ -0,0 +1,49 @@
+import Component from "@ember/component";
+import { observes } from "discourse-common/utils/decorators";
+import { scheduleOnce } from "@ember/runloop";
+
+export default Component.extend({
+  didInsertElement() {
+    this._super(...arguments);
+    this.refreshClass();
+  },
+
+  _updateClass() {
+    if (this.isDestroying || this.isDestroyed) {
+      return;
+    }
+    const topic = this.topic;
```

Comment:
there's not much value in doing this, you can just user `this.topic` everywhere after